FanartHandler/FanartSelectedOther.cs

```diff
@@ -850,6 +855,38 @@ public void AddSelectedMoviePropertys()
         strIMDBID = Utils.GetProperty("#myfilms.db.imdb_id.value");
       }
 
+      var window = GUIWindowManager.GetWindow(Utils.iActiveWindow);
+      GUIListItem selectedListItem = null;
+      if (window != null)
+      {
+        selectedListItem = GUIControl.GetSelectedListItem(Utils.iActiveWindow, window.GetFocusControlId());
+        if (selectedListItem == null && string.IsNullOrEmpty(strIMDBID))
+          return;
+      }

```

Comment:
I don't quite understand why this is. Those. if we don’t have a selected control, and we didn’t get the IMDBID in the code above, then we don’t do anything? Maybe just at the beginning of the procedure, 
https://github.com/andrewjswan/mediaportal-fanart-handler/blob/4d1e117d1046b6fd88977373f195736f387a951f/FanartHandler/FanartSelectedOther.cs#L846-L847
check whether there is an active Control, and if it is not there, just exit without doing anything?

Reply:
This was to try and remove the clearart from Page Navigation on Movies when side panel selected.
The real problem I am trying to address is that `#Trakt.Movie.ImdbId` does not clear on page navigation. For some reason it works for series but not movies. Checking for selected null alone is not good enough because the value is null in MovingPictures.

Reply:
> This was to try and remove the clearart from Page Navigation on Movies when side panel selected.
> The real problem I am trying to address is that `#Trakt.Movie.ImdbId` does not clear on page navigation. For some reason it works for series but not movies.

Then we need to think about how to solve the problem, in principle, in the Media Portal, there is always a selected control.
I understand that Trakt does not clear the property when switching to navigation? In what cases? We seem to have solved this problem.



> Checking for selected null alone is not good enough because the value is null in MovingPictures.

We need to look for a different approach.

Reply:
> Then we need to think about how to solve the problem, in principle, in the Media Portal, there is always a selected control. I understand that Trakt does not clear the property when switching to navigation? In what cases? We seem to have solved this problem.

Only occurs is movies lists. The isFolder check prevents the page navigation from showing an image but because the Trakt value is not cleared the side panel selects the value and shows the clearart.

Reply:
strIMDBID = Utils.GetProperty("#Trakt.Movie.ImdbId");

Reply:
> Only occurs is movies lists. The isFolder check prevents the page navigation from showing an image but because the Trakt value is not cleared the side panel selects the value and shows the clearart.

You can make a helper for Trakt (similar to TV Series) with checks to see if Trakt is installed, whether its window is currently active, and if it is active, then check whether Folders is selected, and if not Folders, then return the desired property. And then just call their helper method.

Reply:
So I tried to check for a null selected list item and give the movie plugins a chance to populate a value before exiting the function

Reply:
> You can make a helper for Trakt (similar to TV Series) with checks to see if Trakt is installed, whether its window is currently active, and if it is active, then check whether Folders is selected, and if not Folders, then return the desired property. And then just call their helper method.

This would move the check we did in Utils for Trakt pages to another class and call it from the AddSelectedMoviePropertys function?

Reply:
I removed the extra code you did not like from AddSelectedMoviePropertys

Reply:
Changed the Utils code back to your original code `if (selectedListItem == null || selectedListItem.IsFolder)` to include the null check. This almost does what we need but it does mean the side menu causes the clearart to disappear because it's selectedListItem is null.

Reply:
Found another issue. While using the TVSeries plugin and selecting artwork or anything to leave the window will cause the clearart to clear and only way to get it back is to select another series.